editor/src/messages/portfolio/portfolio_message_handler.rs

```diff
@@ -426,6 +427,38 @@ impl MessageHandler<PortfolioMessage, PortfolioMessageData<'_>> for PortfolioMes
 				// Upgrade the document's nodes to be compatible with the latest version
 				document_migration_upgrades(&mut document, reset_node_definitions_on_open);
 
+				// Ensure each node has the metadata for its inputs
+				for (node_id, node, path) in document.network_interface.document_network().clone().recursive_nodes() {
+					document.network_interface.validate_input_metadata(node_id, node, &path);
```

Comment:
ensure all input metadata exists for each input, if not then it uses the default from the definition. If there is no definition reference, then it uses an empty name, which will get replaced with the type